### PR TITLE
Make flyway and jdbc-postgresql extensions work without any other required dependencies

### PR DESCRIPTION
**Pull request by geoand**
- Relates to: #41580

### CODE REVIEW

File: extensions/flyway-postgresql/runtime/pom.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    <execution>
+                        <phase>compile</phase>
+                        <goals>
+                            <goal>extension-descriptor</goal>
+                        </goals>
+                        <configuration>
+                            <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}</deployment>
+                            <dependencyCondition>
+                                <artifact>io.quarkus:quarkus-flyway</artifact>
+                            </dependencyCondition>
```
Comment (gastaldi) [inline extensions/flyway-postgresql/runtime/pom.xml:43]: Ah, I missed this setting, that looks cool!

**Review by gastaldi — APPROVED** [review]

**Comment by geoand** [comment]
> But we should probably generalize it to all the supported drivers. Maybe a good first issue for someone?

Yup, I don't plan to do it myself, I made a call [here](https://github.com/quarkusio/quarkus/issues/40834#issuecomment-2200060304)

**Review by aloubyansky — APPROVED** [review]

**Review by FroMage — APPROVED** [review]
So, when I depend on `quarkus-jdbc-postgresql`, there's an optional dep to `quarkus-flyway-postgresql`, which is conditionally upgraded to a full dep if `quarkus-flyway` is also a dependency? Is that how it works?

**Comment by geoand** [comment]
Exactly that!

@aloubyansky had added this feature a whil back, and I've used it in a few places

**Comment by gastaldi** [comment]
Is this feature documented somewhere? That's really cool and I'm thinking of using it in other use cases too

**Comment by geoand** [comment]
Always has been:
https://quarkus.io/guides/conditional-extension-dependencies

😎

**Pull request merged**